appinventor/components/src/com/google/appinventor/components/runtime/ReplForm.java

```diff
@@ -124,20 +124,73 @@ public void onCreate(Bundle icicle) {
     Log.d(LOG_TAG, "onCreate");
     loadedExternalDexs = new ArrayList<String>();
     Intent intent = getIntent();
-    processExtras(intent, false);
+    processExtrasAndData(intent, false);
     themeHelper.setActionBarAnimation(false);
   }
 
   @Override
   void onCreateFinish() {
     super.onCreateFinish();
+    Log.d(LOG_TAG, "onCreateFinish() Called in Repl");
 
     if (!isEmulator() && AppInventorFeatures.doCompanionSplashScreen())
       {                    // Only show REPL splash if not in emulator and enabled
         Intent webviewIntent = new Intent(Intent.ACTION_MAIN);
         webviewIntent.setClassName(activeForm.$context(), SPLASH_ACTIVITY_CLASS);
         activeForm.$context().startActivity(webviewIntent);
       }
+    Intent intent = getIntent();
+    Log.d(LOG_TAG, "Intent = " + intent);
+    final String data = intent.getDataString();
+    if (data != null) {
+      Log.d(LOG_TAG, "Got data = " + data);
+    } else {
+      Log.d(LOG_TAG, "Did not receive any data");
+    }
+
+/////////////////////////////////////////////////////////////////////////
+// Chromebook Support:                                                 //
+//                                                                     //
+// The code below parses the data provided in the intent to get the    //
+// code to use to talk to the rendezvous server. It appears to need to //
+// run on the UI thread (I'm not sure what thread we are on here in    //
+// onCreateFinish().  I'm not sure why we need the delay, but it       //
+// doesn't work if we do not include the delay. I'm continuing to look //
```

Comment:
I looked through the code and haven't been able to figure this out, but maybe another approach would be to register the logic in the runnable as an OnInitialize handler that is run once the REPL has finished initializing--it might be a better choice than trying to tweak the timeout.

Reply:
Thanks for the suggestion. I implemented it and it works great!